mne/io/pick.py

```diff
@@ -784,24 +788,163 @@ def _pick_aux_channels(info, exclude='bads'):
                       ref_meg=False, exclude=exclude)
 
 
-def _pick_data_or_ica(info):
+def _pick_data_or_ica(info, exclude=()):
     """Pick only data or ICA channels."""
     if any(ch_name.startswith('ICA') for ch_name in info['ch_names']):
-        picks = pick_types(info, exclude=[], misc=True)
+        picks = pick_types(info, exclude=exclude, misc=True)
+    else:
+        picks = _pick_data_channels(info, exclude=exclude, with_ref_meg=True)
+    return picks
+
+
+def _picks_to_idx(info, picks, none='data', exclude='bads', allow_empty=False,
+                  with_ref_meg=True):
+    """Convert and check pick validity."""
+    #
+    # None -> all, data, or data_or_ica (ndarray of int)
+    #
+    orig_picks = picks
+    if isinstance(info, int):
+        n_chan = info
+    else:
+        n_chan = info['nchan']
+    assert n_chan >= 0
+
+    if picks is None:
+        if isinstance(info, int):  # special wrapper for no real info
+            picks = np.arange(n_chan)
+        else:
+            picks = none  # let _picks_str_to_idx handle it
+
+    #
+    # slice
+    #
+    if isinstance(picks, slice):
+        picks = np.arange(n_chan)[picks]
+
+    #
+    # -> ndarray of int (and make a copy)
+    #
+    picks = np.atleast_1d(picks)  # this works even for picks == 'something'
+    picks = np.array([], int) if picks.size == 0 else picks
+    if picks.ndim != 1:
+        raise ValueError('picks must be 1D, got %sD' % (picks.ndim,))
+    if picks.dtype.char in ('S', 'U'):
+        picks = _picks_str_to_idx(info, picks, exclude, with_ref_meg)
+    if picks.dtype != np.int:
+        raise TypeError('picks must be a list of int or list of str, got '
+                        'a data type of %s' % (picks.dtype,))
+
+    #
+    # ensure we have (optionally non-empty) ndarray of valid int
+    #
+    if len(picks) == 0 and not allow_empty:
+        raise ValueError('No appropriate channels found for the given picks '
+                         '(%r)' % (orig_picks,))
+    if (picks < -n_chan).any():
+        raise ValueError('All picks must be >= %d, got %r'
+                         % (-n_chan, orig_picks))
+    if (picks >= n_chan).any():
+        raise ValueError('All picks must be < %d, got %r'
```

Comment:
I'd specify that the max is the number of channels for example `... must be < n_channels (%d), ...`